lib/Dialect/TritonGPU/Transforms/WSCodePartition.cpp

```diff
@@ -1319,36 +1608,34 @@ class TritonGPUWSCodePartitionPass
     // channels and an array of buffers for each channel.
     DenseMap<Channel *, DenseMap<int, Value>> barrierAllocMap;
     DenseMap<Channel *, DenseMap<int, Value>> tokenMap =
-        createToken(map, mapKeyVec, funcOp, numConsumerGroups, barrierAllocMap);
+        createToken(channelsGroupedByConsumers, orderedChannels, funcOp,
+                    numConsumerGroups, barrierAllocMap);
     DenseMap<Channel *, Value> bufferMap =
-        createBuffer(channels, funcOp, numConsumerGroups);
+        createBuffer(channelsGroupedByProducers, funcOp, numConsumerGroups);
     LLVM_DEBUG({
       LDBG("\n\nafter createBuffer");
       funcOp.dump();
     });
 
     // Step 6: add async communication ops (ProducerAcquire etc). Also lower the
     // loads.
-    buildAsyncComm(map, tokenMap, barrierAllocMap, bufferMap,
-                   numConsumerGroups);
+    insertAsyncComm(channelsGroupedByConsumers, tokenMap, barrierAllocMap,
+                    bufferMap, numConsumerGroups);
     LLVM_DEBUG({
       LDBG("\n\nwith SyncOps");
       funcOp.dump();
     });
 
+    insertAsyncCopy(funcOp, channelsGroupedByProducers, bufferMap);
```

Comment:
Can we add a comment for this? It seems loads as producers are already lowered in insertAsyncComm.